TFT/src/User/API/LevelingControl.c

```diff
@@ -10,25 +10,31 @@ typedef struct XY_coord
 LEVELING_POINT probedPoint = LEVEL_NO_POINT;  // last probed point or LEVEL_NO_POINT in case of no new updates
 float probedZ = 0.0f;                         // last Z offset measured by probe
 
+int16_t setCoordValue(AXIS axis, ALIGN_POSITION align)
+{
+  return ((align == LEFT || align == BOTTOM) ? infoSettings.machine_size_min[axis] + infoSettings.level_edge
+                                             : infoSettings.machine_size_max[axis] - infoSettings.level_edge) - infoParameters.HomeOffset[axis];
+}
+
 void levelingGetPointCoords(LEVELING_POINT_COORDS coords)
 {
-  int16_t x_left = ((infoSettings.machine_size_min[X_AXIS] < 0) ? 0 : infoSettings.machine_size_min[X_AXIS]) + infoSettings.level_edge;
-  int16_t x_right = infoSettings.machine_size_max[X_AXIS] - infoSettings.level_edge;
-  int16_t y_bottom = ((infoSettings.machine_size_min[Y_AXIS] < 0) ? 0 : infoSettings.machine_size_min[Y_AXIS]) + infoSettings.level_edge;
-  int16_t y_top = infoSettings.machine_size_max[Y_AXIS] - infoSettings.level_edge;
+  int16_t x_left = setCoordValue(X_AXIS, LEFT);
+  int16_t x_right = setCoordValue(X_AXIS, RIGHT);
+  int16_t y_bottom = setCoordValue(Y_AXIS, LEFT);
+  int16_t y_top = setCoordValue(Y_AXIS, RIGHT);
 
   if (GET_BIT(infoSettings.inverted_axis, X_AXIS))
   { // swap left and right
-    int16_t temp = x_left;
-    x_left = x_right;
-    x_right = temp;
+    x_left = x_left ^ x_right;
```

Comment:
Have you disable "Allow edits and access to secrets by maintainers" ?
The fancy usage here is meaningless and only increases the difficulty of understanding. Please restore it

Reply:
Sure, sorry for that.